p4-16/psa/PSA.mdk

```diff
@@ -2066,10 +2068,53 @@ to select a member from the group, and the action specification
 corresponding to that member is applied. The dynamic selection algorithm is
 specified as a parameter when instantiating the action selector.
 
-Action selector members may only specify action types defined
-in the `actions` attribute of the implemented table. All actions in a group must be
-of the same type. The action parameters for actions in the same group are allowed
-to differ, and the action of different groups in a selector may be different.
+Action selector members may only specify action types defined in the
+`actions` attribute of the implemented table.  PSA implementations are
+allowed to restrict all actions in a group to have the same action
+name, but they may allow more general configurations, i.e. different
+action names among members of the same group.  If a PSA implementation
+does have this restriction, it must still allow different members of
+the same group to have different action parameter values.  PSA
+implementations must also allow the action names of different groups
+in a selector to be different.
+
+Consider the situation where controller software wishes to have an
+entry with key `K` in a table with an action selector implementation,
+that points at a group `G` that currently has no members.  This could
+occur because `G` was empty when table entry `(key K, group G)` was
+added, or because `G` formerly had one or more members, and its
+members were then removed.
+
+It is recommended, but not required, that a PSA data plane
+implementation perform the action specified as the value of the
+`psa_empty_group_action` table property if a packet applies the table
+and matches the entry `(K, G)`.  Note that the empty group action
+might have a different action name than `G` contained before, if
+earlier `G` was non-empty.  We will consider requiring such support in
+future versions of PSA.
+
+Until then, implementations are allowed to have non-portable
```

Comment:
Please specify what is supported today first, rather than base it on what will be required in the future. Then say this may be tightened in the future.